### DISCUSSION TITLE
`Error message` 加密后的脚本无法被pyinstaller正确打包其依赖文件

### DISCUSSION THREAD

**Discussion by lingdu0001** [dashingsoft/pyarmor#1231] [Q&A]
Please read this first before report any issue
https://pyarmor.readthedocs.io/en/latest/questions.html

Please copy the whole command line `pyarmor gen` and first 4 lines in the console, do not mask anything, and do not paste snapshot image

```
$ pyarmor gen -O dist --assert-call foo.py
INFO     Python 3.11.3
INFO     Pyarmor 8.1.9 (pro), 005122, non-profits
INFO     Platform windows.x86_64

```
用pyinstaller正常打包（用的spec文件）文件大小为120M。
先用pyarmor  gen -O dist -r src 生成加密文件后
改为用pyinstaller打包加密后的文件，生成的文件大小只有83.1M。
运行文件的时候报错
![image](https://user-images.githubusercontent.com/5213652/237046778-13a35edb-13f4-4314-a709-4bb465ad1a2e.png)

感觉是pyinstaller中少了对应的pip包，用pyinstxtractor反编译以后也发现确实少了pip依赖。

**Comment by jondy** [comment]
请参阅使用文档正确使用 Pyarmor.

**Reply by lingdu0001** [reply]
另外按照文档的简易方法 
pyinstaller foo.py
pyarmor gen --pack dist/foo/foo foo.py
这种方法行不通，因为我是要把一个项目打包为单文件。另外里面需要很多额外的hidden-import.

所以我用的办法是先pyarmor gen -O dist -r src生成加密以后的脚本。然后打补丁给app.spec，在打包app.spec.其中app.spec的入口代码等都指向加密以后的app.py

**Comment by lingdu0001** [comment]
已经按照文档操作了。
先用pyarmor gen -O dist -r src
然后pyinstaller app.spec,其中app.spec按照教程打了补丁

**Reply by 20007883** [reply]
成功了吗？

**Reply by lingdu0001** [reply]
没有

**Comment by 2024baibai** [comment]
--hidden-import flask_migrate
